services/dynamic/json-path.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @module
+ */
+
+'use strict'
+
+const Joi = require('@hapi/joi')
+const jp = require('jsonpath')
+const { renderDynamicBadge, errorMessages } = require('../dynamic-common')
+const { InvalidParameter, InvalidResponse } = require('..')
+
+/**
+ * Dynamic service class factory which wraps {@link module:core/base-service/base~BaseService} with support of {@link https://jsonpath.com/|JSONPath}.
+ *
+ * @param {Function} superclass class to extend
+ * @returns {Function} wrapped class
+ */
+module.exports = superclass =>
```

Comment:
http://justinfagnani.com/2015/12/21/real-mixins-with-javascript-classes/ is a nice resource about "Subclass Factories".